.ci/scripts/test_wheel_package_qnn.sh

```diff
@@ -0,0 +1,155 @@
+
+def main() -> None:
+    install_qnn_sdk()
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", "--output_folder", type=str, default="", help="The folder to store the exported program")
+    parser.add_argument("--soc", type=str, default="SM8650", help="Specify the SoC model.")
+    parser.add_argument("-q", "--quantization", choices=["ptq", "qat"], help="Run post-traininig quantization.")
+    args = parser.parse_args()
+
+    class LinearModule(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.linear = torch.nn.Linear(3, 3)
+        def forward(self, arg):
+            return self.linear(arg)
+        def get_example_inputs(self):
+            return (torch.randn(3, 3),)
+
+    model = LinearModule()
+    example_inputs = model.get_example_inputs()
+
+    if args.quantization:
+        quantizer = QnnQuantizer()
+        m = torch.export.export(model.eval(), example_inputs, strict=True).module()
+        if args.quantization == "qat":
+            m = prepare_qat_pt2e(m, quantizer)
+            m(*example_inputs)
+        elif args.quantization == "ptq":
+            m = prepare_pt2e(m, quantizer)
+            m(*example_inputs)
+        m = convert_pt2e(m)
+    else:
+        m = model
+
+    use_fp16 = True if args.quantization is None else False
+    backend_options = generate_htp_compiler_spec(use_fp16=use_fp16)
+    compile_spec = generate_qnn_executorch_compiler_spec(
+        soc_model=get_soc_to_chipset_map()[args.soc],
+        backend_options=backend_options,
+    )
+    delegated_program = to_edge_transform_and_lower_to_qnn(m, example_inputs, compile_spec)
+    executorch_program = delegated_program.to_executorch(
+        config=ExecutorchBackendConfig(extract_delegate_segments=False)
+    )
+    save_pte_program(executorch_program, "linear", args.output_folder)
+
+if __name__ == "__main__":
+    main()
+EOF
+
+# ----------------------------
+# Wheel build and .so checks
+# ----------------------------
+echo "=== Building Wheel Package ==="
+EXECUTORCH_BUILDING_WHEEL=1 python setup.py bdist_wheel
+
+unset EXECUTORCH_BUILDING_WHEEL
+
+WHEEL_FILE=$(ls dist/*.whl | head -n 1)
+echo "Found wheel: $WHEEL_FILE"
+
+# ----------------------------
+# Check wheel does NOT contain qualcomm/sdk
+# ----------------------------
+echo "Checking wheel does not contain qualcomm/sdk..."
+SDK_FILES=$(unzip -l "$WHEEL_FILE" | awk '{print $4}' | grep "executorch/backends/qualcomm/sdk" || true)
+if [ -n "$SDK_FILES" ]; then
+    echo "ERROR: Wheel package contains unexpected qualcomm/sdk files:"
+    echo "$SDK_FILES"
+    exit 1
+else
+    echo "OK: No qualcomm/sdk files found in wheel"
+fi
+
+# ----------------------------
+# Conda environment setup & tests
+# ----------------------------
+TEMP_ENV_DIR=$(mktemp -d)
+echo "Using temporary directory: $TEMP_ENV_DIR"
+
+conda create -y -p "$TEMP_ENV_DIR/env" python=3.10
```

Comment:
should we also check for other python versions? Looks like we do both 3.10 and 3.12 https://pypi.org/project/executorch/0.7.0/#files, so maybe test both?

Reply:
Also if venv is used, can we check it works with venv as well?